// fun-util-fun-v1/src/index.js
const { EROFS, PRIORITY_ABOVE_NORMAL } = require('constants')
const { get } = require('http')

module.exports ={
    calculator: function(num1,sum,num2){
        if(!num1) return new TypeError('Num1 has no value')
        if(!sum) return new TypeError('Invalid Operator/sum type.')
        if(!num2) return new TypeError('Missing input of num2')
        if(sum === '*') return num1*num2
        if(sum === '-') return num1-num2
        if(sum === '+') return num1+num2
        if(sum === '/') return num1/num2
    },
    getdiscordbot: function(token){
        const fs = require('fs')
        if(!token) return new TypeError('No token specified')
            const Discord = require('discord.js')
            if(!Discord) return new TypeError('Please install discord.js')
            const client = new Discord.Client()
            const token1 = token;
            client.on('ready', () => {
                console.log(`Hi, ${client.user.username} is now online! (no commands)`)
            })
            client.login(token1)

        },
        storedata: function(dataname, content){
            const fs = require('fs')
            if(!dataname) return new TypeError('Did not specify name for data')
            if(!content) return new TypeError('Did not specify content')
            if(dataname, content){
                fs.writeFileSync(`${dataname}.txt`, `${content}`)
             console.log(`Name of: ${dataname} with content of: ${content} stored.`)
             console.log("if the dataname has already been stored, it will overwrite content.")
             
            }
            },
            error: function(error){
                const fs = require('fs')
              if(!error) return new TypeError("Give an error to emit the error")
                if(error){
                    fs.writeFileSync('fun-util-fun-v1-errors.md', `Oops, we had an error with fun-util-fun-v1!! Please tell the error below to support! Error: ${error}\nIf any of this errors was fake, and was reported to the developer. We will report you.`)
                    console.log('Wrote Error!')
                }
            },
            print: function(print){
                if(!print) return console.log('No print')
                if(print){
                    console.log(print)
                }
            },
            p: function(print){
                if(!print) return console.log('No print display')
                if(print){
                    console.log(print)
                }
            },
            support: function(getsupport, name, email){
                const fetch = require('node-fetch')
                const config = require('./extra/webhooks/webhooks')
                // make sure you are in the discord support
                if(!getsupport) return console.log('[first arg]You did not specify what you need help with!')
                if(!name) return console.log('[2nd arg] You did not specify your name')
                if(!email) return console.log("[3rd arg // last arg] did not specify email.")
                if(getsupport){
                    const whurl = config.web1;
                    const msgjson = {
                        "content": ` ${name} needs help with ${getsupport} their email is: ${email}`
                    }
                    fetch(whurl, {"method": "POST", "headers": {"content-type": "application/json"},

                    "body": JSON.stringify(msgjson)})
                    console.log('We have contacted our moderators for support... Please do not spam us or we will report you. We will send you an email.')
                    console.log('Notice: if you put any fake email, we will have no way to contact you for support.')
                }
            },
            suggest: function(suggestion, name){
                if(!suggestion) return console.log('Did not specify a suggestion')
                if(!name) return console.log('Did not specify a name')
                if(suggestion){
                    const config = require('./extra/webhooks/webhooks')
                    const fetch = require('node-fetch')
                    const whurl = config.web2;
                    const m = {
                        "content": `${name} suggested ${suggestion}`
                    }
                    fetch(whurl, {"method": "POST", "headers": {"content-type": "application/json"},

                    "body": JSON.stringify(m)})
                    console.log('We have recorded your suggestion.')
                    console.log('Please do not spam your suggestion in the system or we will report you')
                }
            },
            webhook: function(webhook, content){
                const fs = require('fs')
                if(!webhook) return console.log('Did not specify a webhook.')
                if(!content) return console.log('Did not specify your content')
                if(webhook, content){
                    const fetch = require('node-fetch')
                    const whurl = webhook;
                    const m = {
                        "content": `${content}`
                    }
                    fs.writeFileSync('webhooklatest.js', `const latest ='${webhook}'\n // this is used for if you want to use a webhook as a variable`)
                    fetch(whurl, {"method": "POST", "headers": {"content-type": "application/json"},

                    "body": JSON.stringify(m)})
                    console.log('The content has been sent on the webhook.')
                    console.log('If it was a fake webhook nothing will really happen')
                }
            }
        }
